src/Controller/AuthorizationController.php

```diff
@@ -56,27 +57,43 @@ public function __construct(
      */
     public function __invoke(ServerRequestInterface $request): ResponseInterface
     {
-        $authorizationRequest = $this->authorizationServer->validateAuthorizationRequest($request);
+        $queryParameters = $request->getQueryParams();
+        if (!isset($queryParameters[ProcessingChain::AUTHPARAM])) {
+            $authorizationRequest = $this->authorizationServer->validateAuthorizationRequest($request);
+            $this->authenticationService->processRequest($request, $authorizationRequest);
```

Comment:
My preference would be for `processRequest` to not redirect unless one of the authproc filters issues one. I think you can achieve that by having it return the $state array, not call ProcessingComplete and then put ` $state = $this->authenticationService->loadState($queryParameters);` into the else.

My reason for preferring this is that the vast majority of our logins do not need to redirect the user, and adding the extra redirect makes the system feel slower.

Reply:
done